src/daemon/custom_image_host.cpp

```diff
@@ -20,64 +20,195 @@
+        if (line.endsWith(image_file.toUtf8()))
         {
             hash = QString(line.split(' ').first());
             break;
         }
     }
 
-    mp::VMImageInfo core_image_info{
-        {"core"}, "core-16", "Core 16", true, url, "", "", hash, last_modified.toString("yyyyMMdd"), 0};
+    return BaseImageInfo{last_modified, hash};
+}
+
+auto map_aliases_to_vm_info_for(const std::vector<mp::VMImageInfo>& images)
+{
+    std::unordered_map<std::string, const mp::VMImageInfo*> map;
+    for (const auto& image : images)
+    {
+        map[image.id.toStdString()] = &image;
+        for (const auto& alias : image.aliases)
+        {
+            map[alias.toStdString()] = &image;
+        }
+    }
+
+    return map;
+}
+
+auto multipass_default_aliases(mp::URLDownloader* url_downloader, const QString& path_prefix)
+{
+    std::vector<mp::VMImageInfo> default_images;
+    QString image_url;
+    QString hash_url;
+    QString image_file{"ubuntu-core-16-amd64.img.xz"};
+
+    if (!path_prefix.isEmpty())
+    {
+        image_url = QUrl::fromLocalFile(path_prefix + image_file).toString();
+        hash_url = QUrl::fromLocalFile(path_prefix + "SHA256SUMS").toString();
+    }
+    else
+    {
+        image_url = "http://cdimage.ubuntu.com/ubuntu-core/16/current/" + image_file;
+        hash_url = "http://cdimage.ubuntu.com/ubuntu-core/16/current/SHA256SUMS";
+    }
+
+    auto base_image_info = base_image_info_for(url_downloader, image_url, hash_url, image_file);
+    mp::VMImageInfo core_image_info{{"core"},
+                                    "Ubuntu",
+                                    "core-16",
+                                    "Core 16",
+                                    true,
+                                    image_url,
+                                    "",
+                                    "",
+                                    base_image_info.hash,
+                                    base_image_info.last_modified,
+                                    0};
+
+    default_images.push_back(core_image_info);
 
-    default_aliases.insert({"core", core_image_info});
+    auto map = map_aliases_to_vm_info_for(default_images);
 
-    return default_aliases;
+    return std::unique_ptr<mp::CustomManifest>(new mp::CustomManifest{std::move(default_images), std::move(map)});
+}
+
+auto snapcraft_default_aliases(mp::URLDownloader* url_downloader, const QString& path_prefix)
+{
+    std::vector<mp::VMImageInfo> default_images;
+    QString image_url;
+    QString hash_url;
+    QString image_file;
+
+    {
+        image_file = "ubuntu-16.04-server-cloudimg-amd64-disk1.img";
+        if (!path_prefix.isEmpty())
+        {
+            image_url = QUrl::fromLocalFile(path_prefix + image_file).toString();
+            hash_url = QUrl::fromLocalFile(path_prefix + "SHA256SUMS").toString();
+        }
+        else
+        {
+            image_url = "https://cloud-images.ubuntu.com/releases/16.04/release/" + image_file;
```

Comment:
mind changing this to http://cloud-images.ubuntu.com/minimal/releases/xenial/release/